extensions/markdown-language-features/src/features/workspaceSymbolProvider.ts

```diff
@@ -81,8 +90,23 @@ class VSCodeWorkspaceMarkdownDocumentProvider extends Disposable implements Work
 	}
 
 	private async getMarkdownDocument(resource: vscode.Uri): Promise<SkinnyTextDocument | undefined> {
-		const doc = await vscode.workspace.openTextDocument(resource);
-		return doc && isMarkdownFile(doc) ? doc : undefined;
+		let result = this._docIndex.getByUri(resource);
```

Comment:
nits:
1) rename to `existingDocument`.
2) since we already have the document in memory, can we just return it? i.e. does `TextDocument` cover `SkinnyTextDocument`?